### PR TITLE
adds model and references

### PR DESCRIPTION
**Pull request by cubadomingo**
Couple of things I did here:

- added city_id foreign key to **events**

- created a **subregion** model with a region_id foreign key

- added category_id foreign key to **post** and **events**

- added user_id, event_id, and post_id foreign key to **comments**

### CODE REVIEW

**Comment by cubadomingo** [comment]
Take a look at it when you can and if everything looks good, go ahead and merge it.

**Comment by tmobaird** [comment]
it looks good to me so far, I wonder why there are so many lines like `t.index ["subregion_id"], name: "index_cities_on_subregion_id", using: :btree` in the schema file. Looks good to me though

**Comment by cubadomingo** [comment]
I basically added all the missing references through the generate migrate command like such:

 `rails g migrate AddCitiesToEvents city:references`

This adds the line you're referring to in the schema. I'll break it down:

`t.index ["subregion_id"]`

 adds the foreign key

`name: "index_cities_on_subregion_id"`

 just gives the foreign key index a name for making it easier to refer to. Say if we were doing raw SQL what it there was more than two subregion foreign keys? The name should tell you which is which.

`using: :btree`

This just tells active model to use a B Tree which is a data structure for indexing. Here's a eli5 because I don't really understand how the magic behind it works.

screen shot 2016-07-29 at 7 04 35 pm

**Comment by tmobaird** [comment]
Roger that dude. thanks for adding all this! :+1: I'm gonna get started on the feed and react components tonight

**Comment by cubadomingo** [comment]
💯💯💯
